app/src/main/java/org/oppia/android/app/topic/conceptcard/ConceptCardFragment.kt

```diff
@@ -22,23 +23,74 @@ class ConceptCardFragment : InjectableDialogFragment() {
 
   companion object {
     /** The fragment tag corresponding to the concept card dialog fragment. */
-    const val CONCEPT_CARD_DIALOG_FRAGMENT_TAG = "CONCEPT_CARD_FRAGMENT"
+    private const val CONCEPT_CARD_DIALOG_FRAGMENT_TAG = "CONCEPT_CARD_FRAGMENT"
 
-    /**
-     * Creates a new fragment to show a concept card.
-     *
-     * @param skillId the skill ID for which a concept card should be loaded
-     * @param profileId the profile in which the concept card will be shown
-     * @return a new [ConceptCardFragment] to display the specified concept card
-     */
-    fun newInstance(skillId: String, profileId: ProfileId): ConceptCardFragment {
+    private fun newInstance(skillId: String, profileId: ProfileId): ConceptCardFragment {
       return ConceptCardFragment().apply {
         arguments = Bundle().apply {
           putString(SKILL_ID_ARGUMENT_KEY, skillId)
           putProto(PROFILE_ID_ARGUMENT_KEY, profileId)
         }
       }
     }
+
+    /**
+     * Removes any [ConceptCardFragment] in the given FragmentManager that is not of the given
+     * skill Id. If no [ConceptCardFragment] remains, creates a new fragment to show a concept card
+     * for the given skill id.
+     *
+     * @param skillId the skill ID for which a concept card should be loaded.
+     * @param profileId the profile in which the concept card will be shown.
+     * @param fragmentManager the [FragmentManager] where to show the concept card.
+     * @return the fragment created or null if none.
+     */
+    fun bringToFrontOrCreateIfNew(
+      skillId: String,
+      profileId: ProfileId,
+      fragmentManager: FragmentManager
+    ): ConceptCardFragment? {
+      // The UI is recreated when changing profile ids, so no need to include it in the tag name.
```

Comment:
This comment seems a bit out-of-band--perhaps add more context to explain the potential issue & why this means it isn't a problem?

Reply:
Expanded the comment. I could also complicate the code so that the comment is not necessary.

Reply:
I think the new comment is much clearer now--thanks!